File: 217. Contains Duplicate.go
package main

import (
	"fmt"
)

func containsDuplicate(nums []int) bool {
	set := map[int]bool{}
	for i := 0; i < len(nums); i++ {
		if _, didFind := set[nums[i]]; didFind {
			return true
		} else {
			set[nums[i]] = true
		}
	}
	return false
}

func main() {

	input := []int{1, 2, 3, 1}

	fmt.Println(input)

}

File: 217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        # Solution 1 - Sorting
        # O(1) - Space
        # O(nlogn) - Time

        nums.sort()

        for index in range(len(nums)):
            if index == len(nums) - 1: return False
            if nums[index] == nums[index+1]: return True


        # Solution 2 - Hashtable
        # O(n) - Space
        # O(n) - Time

        hashset = set()

        for num in nums:
            if num in hashset: return True
            else: hashset.add(num)

        return False




